eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/Binary/CostModelsSpec.hs

```diff
@@ -9,30 +10,43 @@
 module Test.Cardano.Ledger.Alonzo.Binary.CostModelsSpec (spec) where
 
 import Cardano.Ledger.Alonzo.Core
-import Cardano.Ledger.Alonzo.Scripts
 import Cardano.Ledger.BaseTypes
 import Cardano.Ledger.Binary (ToCBOR (..))
 import Cardano.Ledger.Binary.Decoding
 import Cardano.Ledger.Binary.Encoding
+import Cardano.Ledger.Plutus.CostModels
 import Cardano.Ledger.Plutus.Language
 import qualified Data.Map as Map
+import Data.Proxy
 import Data.Text (Text)
-import Data.Typeable
 import Data.Word (Word8)
 import Lens.Micro
 import Test.Cardano.Ledger.Alonzo.Arbitrary
 import Test.Cardano.Ledger.Common
 
-spec ::
-  forall era.
-  AlonzoEraPParams era =>
-  Proxy era ->
-  Spec
-spec pxy = do
-  describe "CostModels CBOR deserialization" $ do
-    validCostModelProp @era pxy
+spec :: forall era. AlonzoEraPParams era => Spec
+spec = do
+  describe "CBOR deserialization" $ do
+    validCostModelProp @era Proxy
     invalidCostModelProp @era Proxy
     unknownCostModelProp @era Proxy
+  prop "applyPPUpdates" $ \valid validUpdate unknown unknownUpdate -> do
+    let
+      validExpected = mkCostModels (costModelsValid validUpdate <> costModelsValid valid)
+      unknownExpected = unknownUpdate <> unknown
```

Comment:
I suppose the chance of the generated `unknown` to be actually a valid one can be neglected

Reply:
On the contrary, it is desired. I actually found a bug in my implementation earlier thanks to the fact that we generate unknown updates that can be valid

Reply:
I see, because we're flattening them after. Makes sense.